evm_arithmetization/src/cpu/kernel/interpreter.rs

```diff
@@ -506,17 +402,17 @@ impl<'a, F: Field> Interpreter<'a, F> {
         Ok(())
     }
 
-    fn code(&self) -> &MemorySegmentState {
+    pub(crate) fn code(&self) -> &MemorySegmentState {
         // The context is 0 if we are in kernel mode.
         &self.generation_state.memory.contexts[(1 - self.is_kernel() as usize) * self.context()]
             .segments[Segment::Code.unscale()]
     }
 
     fn code_slice(&self, n: usize) -> Vec<u8> {
         let pc = self.generation_state.registers.program_counter;
-        self.code().content[pc..pc + n]
+        self.code().content[pc + 1..pc + n + 1]
```

Comment:
Why the `+ 1`? How come we need to shift the content one step?

Reply:
The interpreter's program counter was actually equal to `program_counter + 1` before the unification, hence why we didn't have the +1 before.